src/anomalib/data/utils/augmenter.py

```diff
@@ -168,3 +170,75 @@ def augment_batch(self, batch: Tensor) -> tuple[Tensor, Tensor]:
         augmented_batch = batch * (1 - masks) + (beta) * perturbations + (1 - beta) * batch * (masks)
 
         return augmented_batch, masks
+
+
+class PerlinROIAugmenter(Augmenter):
+    def augment_batch(self, batch: Tensor, thresh_batch: Tensor | None = None) -> tuple[Tensor, Tensor]:
```

Comment:
Can you add a short description on how this works. Something like, extracts foreground object based on binary thresholding and limits perlin noise augmentation to that region.

Reply:
added as the docstring for this class